src/atomate2/abinit/flows/dfpt.py

```diff
@@ -0,0 +1,265 @@
+from jobflow import Flow, Maker
+
+from atomate2.abinit.jobs.anaddb import AnaddbDfptDteMaker, AnaddbMaker
+from atomate2.abinit.jobs.core import StaticMaker
+from atomate2.abinit.jobs.mrgddb import MrgddbMaker
+from atomate2.abinit.jobs.response import (
+    DdeMaker,
+    DdkMaker,
+    DteMaker,
+    generate_dde_perts,
+    generate_ddk_perts,
+    generate_dte_perts,
+    run_rf,
+)
+from atomate2.abinit.powerups import update_factory_kwargs, update_user_abinit_settings
+from atomate2.abinit.sets.core import StaticSetGenerator
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from pymatgen.core.structure import Structure
+
+    from atomate2.abinit.jobs.base import BaseAbinitMaker
+
+
+@dataclass
+class DfptFlowMaker(Maker):
+    """
+    Maker to generate a DFPT flow with abinit.
+
+    The classmethods allow to tailor the flow for specific properties
+        accessible via DFPT.
+
+    Parameters
+    ----------
+    name : str
+        Name of the flows produced by this maker.
+    static_maker : .BaseAbinitMaker
+        The maker to use for the static calculation.
+    ddk_maker : .BaseAbinitMaker
+        The maker to use for the DDK calculations.
+    dde_maker : .BaseAbinitMaker
+        The maker to use for the DDE calculations.
+    dte_maker : .BaseAbinitMaker
+        The maker to use for the DTE calculations.
+    mrgddb_maker : .Maker
+        The maker to merge the DDE and DTE DDB.
+    use_ddk_sym : bool
+        True if only the irreducible DDK perturbations should be considered,
+            False otherwise.
+    use_dde_sym : bool
+        True if only the irreducible DDE perturbations should be considered,
+            False otherwise.
+    dte_skip_permutations: Since the current version of abinit always performs
+        all the permutations of the perturbations, even if only one is asked,
+        if True avoids the creation of inputs that will produce duplicated outputs.
+    dte_phonon_pert: is True also the phonon perturbations will be considered.
+        Default False.
+    dte_ixc: Value of ixc variable. Used to overwrite the default value read
+        from pseudos.
+    """
+
+    name: str = "DFPT"
+    static_maker: BaseAbinitMaker = field(
+        default_factory=lambda: StaticMaker(
+            input_set_generator=StaticSetGenerator(factory=scf_for_phonons)
+        )
+    )
+    ddk_maker: BaseAbinitMaker | None = field(default_factory=DdkMaker)  # |
+    dde_maker: BaseAbinitMaker | None = field(
+        default_factory=DdeMaker
+    )  # | VT: replace by bool?
+    dte_maker: BaseAbinitMaker | None = field(default_factory=DteMaker)  # |
+    mrgddb_maker: Maker | None = field(default_factory=MrgddbMaker)  # |
+    anaddb_maker: Maker | None = field(default_factory=AnaddbMaker)  # |
+    use_ddk_sym: bool | None = False
+    use_dde_sym: bool | None = False
+    dte_skip_permutations: bool | None = False
+    dte_phonon_pert: bool | None = False
+    dte_ixc: int | None = None
```

Comment:
In the end I removed it because it was only affecting the generation of the perturbations, not the calculation in itself. Can be added somewhere in the future if needed.